src/main/scala/org/clulab/wm/eidos/context/GeoNorm.scala

```diff
@@ -26,36 +26,58 @@ object GeoNormFinder {
 
   private lazy val logger = LoggerFactory.getLogger(getClass)
 
-  def fromConfig(config: Config): GeoNormFinder = {
-    val geoNamesIndexPath = Paths.get(config[String]("geoNamesIndexPath"))
-    val geoNamesIndexURL: URL = config[URL]("geoNamesIndexURL")
-
-    // if the index is not yet local, download it
-    if (!Files.exists(geoNamesIndexPath.resolve("segments_1"))) {
-      logger.info(s"No GeoNames index at $geoNamesIndexPath.")
-
+  class CacheManager(config: Config) {
+    lazy val geoNamesIndexPath: Path = Paths.get(config[String]("geoNamesIndexPath"))
+    protected lazy val segmentsPath = geoNamesIndexPath.resolve("segments_1")
+    protected lazy val zipPath = geoNamesIndexPath.resolve("geonames-index.zip")
+
+    // The default is not to replace any files on a machine that is simply running Eidos.
+    // This can be overruled by programs that are managing the cache.
+    def mkCache(replaceOnUnzip: Boolean = false): Unit = {
       // copy the zip file to the local machine
+      val geoNamesIndexURL: URL = config[URL]("geoNamesIndexURL")
       logger.info(s"Downloading the GeoNames index from $geoNamesIndexURL.")
-      val zipPath = geoNamesIndexPath.resolve("geonames-index.zip")
       Files.createDirectories(geoNamesIndexPath)
       Files.copy(geoNamesIndexURL.openStream, zipPath)
 
       // unzip the zip file
       logger.info(s"Extracting the GeoNames index to $geoNamesIndexPath.")
-      FileUtils.unzip(zipPath, geoNamesIndexPath)
-
-      // remove the downloaded zip file
+      FileUtils.unzip(zipPath, geoNamesIndexPath, replace = replaceOnUnzip)
       Files.delete(zipPath)
+
+      if (!isCached)
+        throw new RuntimeException(s"The caching operation was apparently unsuccessful.")
+    }
+
+    def rmCache(): Unit = {
+      Files.deleteIfExists(segmentsPath)
+      Files.deleteIfExists(zipPath)
     }
 
+    def isCached: Boolean = {
+      val cached = Files.exists(segmentsPath)
+
+      if (cached)
+        logger.info(s"No GeoNames index at $geoNamesIndexPath.")
+      else
+        logger.info(s"GeoNames index found at $geoNamesIndexPath.")
```

Comment:
I believe these are backwards. (Only noticed it after watching the log while testing other GeoNames fixes.)

Reply:
Thanks!